tests/libs/nil_core/test_block.cpp

```diff
@@ -0,0 +1,55 @@
+#include <cstddef>
+#include <string>
+#include <vector>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "ssz++.hpp"
+#include "zkevm_framework/core/types/block.hpp"
+
+std::vector<std::byte> hex_to_bytes(const std::string& hex) {
+    std::vector<std::byte> bytes;
+
+    for (unsigned int i = 0; i < hex.length(); i += 2) {
+        std::string byteString = hex.substr(i, 2);
+        unsigned char byte = (unsigned char)strtol(byteString.c_str(), NULL, 16);
+        bytes.push_back(std::byte{byte});
+    }
+
+    return bytes;
+}
+
+// This is a bad test which needs to be replaces with proper integration testing:
+// load ssz-serialized entity from nil and pass it to NilCore.
+
+TEST(NilCoreBlockTests, SerializeDeserializeBlock) {
+    const std::string hex_ssz =
```

Comment:
Let's avoid using predefined serialized values. It should be check on the separate integration tests.
I guess we can make easy support of unit test by using template function:
```
template <typename T>
test_serialization(const T& obj) {
    auto bytes = ssz::serialize(obj);
    auto new_obj = ssz::deserialize<T>(bytes);
    EXPECT_TRUE(compare<T>(new_obj, obj));
}
```

Reply:
and merge all tests into one file

Reply:
Yeah, I added it just to check at least something before proper testing with `nil`. There is actually not much that we can test here in unit tests.